core/workspace_svg.js

```diff
@@ -732,6 +803,87 @@ Blockly.WorkspaceSvg.prototype.markFocused = function() {
   Blockly.mainWorkspace = this;
 };
 
+/**
+ * Zooming the blocks centered in (x, y) coordinate with zooming in or out.
+ * @param {!number} x X coordinate of center.
```

Comment:
Primitives are not nullable by default, so just {number}.

To clarify in general:
{Foo} -> Foo object or null
{!Foo} -> Foo object
{string} -> string primitive (not a String)
{?string} -> string primitive or null